components/table/table-wrapper.js

```diff
@@ -0,0 +1,318 @@
+		background-color: inherit; /* white background so sticky cells layer on top of non-sticky cells */
+	}
+	d2l-table-wrapper[sticky-headers] .d2l-table > tbody > tr:not([header]) > [sticky] {
+		background-color: inherit;
+		left: 0;
+		position: -webkit-sticky;
+		position: sticky;
+		z-index: 1;
+	}
+`;
+
+/**
+ *
+ * Wraps a native <table> element, providing styling and scroll buttons for overflow.
+ *
+ * @slot - Content to wrap
+ */
+export class TableWrapper extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether header row is sticky
+			 */
+			stickyHeaders: {
+				attribute: 'sticky-headers',
+				reflect: true,
+				type: Boolean
+			},
+			/**
+			 * Type of table style to apply
+			 * @type {'default'|'light'}
+			 */
+			type: {
+				reflect: true,
+				type: String
+			}
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				--d2l-table-border: 1px solid var(--d2l-table-border-color);
+				--d2l-table-border-color: var(--d2l-color-mica);
+				--d2l-table-border-radius: 0.3rem;
+				--d2l-table-cell-height: 41px; /* min-height to be 62px including border */
+				--d2l-table-cell-padding: 0.5rem 1rem;
+				--d2l-table-header-background-color: var(--d2l-color-regolith);
+				--d2l-table-row-border-color-selected: var(--d2l-color-celestine);
+				--d2l-table-row-background-color-selected: var(--d2l-color-celestine-plus-2);
+				display: block;
+				width: 100%;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			:host([type="light"]) {
+				--d2l-table-cell-height: 1.15rem; /* min-height to be 48px including border */
+				--d2l-table-cell-padding: 0.6rem;
+				--d2l-table-border-color: var(--d2l-color-gypsum);
+				--d2l-table-header-background-color: #ffffff;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.stickyHeaders = false;
+		this.type = 'default';
+		this._tableObserver = null;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._tableObserver) this._tableObserver.disconnect();
+	}
+
+	render() {
+		const slot = html`<slot @slotchange="${this._handleSlotChange}"></slot>`;
+		if (this.stickyHeaders) {
+			return slot;
+		} else {
+			return html`<d2l-scroll-wrapper>${slot}</d2l-scroll-wrapper>`;
+		}
+	}
+
+	updated(changedProperties) {
+		// hack: grades/groups/outcomes in the LE use this CSS class on the
+		// body to apply special CSS to the page when tables are sticky
+		// Ideally they should be adding this class to the body.
+		if (changedProperties.has('stickyHeaders')) {
+			if (this.stickyHeaders) {
+				document.body.classList.add('d2l-table-sticky-headers');
+			}
+		}
+	}
+
+	async _applyClassNames(table) {
+
+		await new Promise((resolve) => requestAnimationFrame(() => resolve()));
+
+		// offsetParent causes reflow/paint so do them all at once
+		const rows = Array.from(table.rows);
+		let firstRow = null;
+		let lastRow = null;
+		rows.forEach((r) => {
```

Comment:
Nit: this could become:
```javascript
rows.forEach((r) => {
	if (r.offsetParent === null) return;
	if (firstRow === null) {
		firstRow = r;
	}
	lastRow = r;
});
```